runtime/vm/ClassInitialization.cpp

```diff
@@ -386,6 +386,22 @@ classInitStateMachine(J9VMThread *currentThread, J9Class *clazz, J9ClassInitStat
 				if (NULL == initializationLock) {
 					goto done;
 				}
+#if defined(J9VM_OPT_VALHALLA_VALUE_TYPES)
+				PUSH_OBJECT_IN_SPECIAL_FRAME(currentThread, initializationLock);
+				/* Preparation is the earliest point where the defaultValue would needed. 
+				 * I.e pre-filling static fields. Therefore, the defaultValue must be allocated at 
+				 * the end of verification 
+				 */
+				j9object_t defaultValue = vm->memoryManagerFunctions->J9AllocateObject(currentThread, clazz, J9_GC_ALLOCATE_OBJECT_TENURED | J9_GC_ALLOCATE_OBJECT_NON_INSTRUMENTABLE);
+				initializationLock = POP_OBJECT_IN_SPECIAL_FRAME(currentThread);
+				clazz = VM_VMHelpers::currentClass(clazz);
+				if (NULL == defaultValue) {
+					setHeapOutOfMemoryError(currentThread);
+					goto done;
+				}
+				J9FlattenedClassCacheHeader *flattenedClassCacheHead = (J9FlattenedClassCacheHeader *) clazz->flattenedClassCache;
+				flattenedClassCacheHead->defaultValue = defaultValue;
```

Comment:
@dmitripivkine @amicic Is a barrier required here?

Reply:
Yes (J9STATIC_OBJECT_STORE)

Reply:
@gacholio @amicic Please review. Is this the correct way to call the barrier?

J9STATIC_OBJECT_STORE(currentThread, clazz, flattenedClassCacheHeader->defaultValue, defaultValue);

Reply:
Need to pass the address of the destination -  missing & operator.